devices/Ft6xx6x/samples/Program.cs

```diff
@@ -1,55 +1,226 @@
-using I2cDevice device = I2cDevice.Create(settings);
-using GpioController gpio = new();
-
-using Ft6xx6x sensor = new(device);
-var ver = sensor.GetVersion();
-Debug.WriteLine($"version: {ver}");
-sensor.SetInterruptMode(false);
-Debug.WriteLine($"Period active: {sensor.PeriodActive}");
-Debug.WriteLine($"Period active in monitor mode: {sensor.MonitorModePeriodActive}");
-Debug.WriteLine($"Time to enter monitor: {sensor.MonitorModeDelaySeconds} seconds");
-Debug.WriteLine($"Monitor mode: {sensor.MonitorModeEnabled}");
-Debug.WriteLine($"Proximity sensing: {sensor.ProximitySensingEnabled}");
-
-gpio.OpenPin(39, PinMode.Input);
-// This will enable an event on GPIO39 on falling edge when the screen if touched
-gpio.RegisterCallbackForPinValueChangedEvent(39, PinEventTypes.Falling, TouchInterrupCallback);
-
-while (true)
+namespace NFAppCore2App2
 {
-    Thread.Sleep(20);
-}
+    public class Program
+    {
+        public static void Main()
+        {
+            // Note: this sample requires a M5Core2.
+            // If you want to use another device, just remove all the related nugets
+            // And comments the following line
+            // You will need as well to set the pins if needed.
+            M5Core2.InitializeScreen();
+            nanoFramework.Console.Clear();
+            nanoFramework.Console.WriteLine("Core2 Touch screen test");
+            nanoFramework.Console.WriteLine("Touch Chip type: " + Iot.Device.Ft6xx6x.ChipType.Ft6336U.ToString());
 
-void TouchInterrupCallback(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
-{
-    Debug.WriteLine("Touch interrupt");
-    var points = sensor.GetNumberPoints();
-    if (points == 1)
+            I2cConnectionSettings settings = new(1, Ft6xx6x.DefaultI2cAddress);
+            using I2cDevice device = I2cDevice.Create(settings);
+            using GpioController gpio = new();
+
+            using Ft6xx6x sensor = new(device);
+            var ver = sensor.GetVersion();
+            Debug.WriteLine($"version: {ver}");
+            sensor.SetInterruptMode(false);
+            Debug.WriteLine($"Period active: {sensor.PeriodActive}");
+            Debug.WriteLine($"Period active in monitor mode: {sensor.MonitorModePeriodActive}");
+            Debug.WriteLine($"Time to enter monitor: {sensor.MonitorModeDelaySeconds} seconds");
+            Debug.WriteLine($"Monitor mode: {sensor.MonitorModeEnabled}");
+            Debug.WriteLine($"Proximity sensing: {sensor.ProximitySensingEnabled}");
+
+            gpio.OpenPin(39, PinMode.Input);
+            // This will enable an event on GPIO39 on falling edge when the screen if touched
+            gpio.RegisterCallbackForPinValueChangedEvent(39, PinEventTypes.Falling, TouchInterrupCallback);
+
+            touchPanelUserAppTemplate touchPanelUserApp = new touchPanelUserAppTemplate();
+            TouchPanelProcessor.TouchedEvent += touchPanelUserApp.userDefinedTouchPanelBehaviourSkeleton;
+
+            while (true)
```

Comment:
If nothing else it's happening in Main() this should be thread.sleep(infinite) otherwise it will waste cpu cycles doing nothing and delaying other threads

Reply:
that one is not on my shoulders, it was on original code from Ellenbach https://github.com/nanoframework/nanoFramework.IoT.Device/blob/develop/devices/Ft6xx6x/samples/Program.cs ...but i will improve it...